evm_arithmetization/src/keccak/round_flags.rs

```diff
@@ -24,10 +24,20 @@ pub(crate) fn eval_round_flags<F: Field, P: PackedField<Scalar = F>>(
         yield_constr.constraint(current_round_flag * (current_round_flag - F::ONE));
     }
 
+    // Constrain the flags to be either 0 or 1.
```

Comment:
Regarding the first loop: sorry, I think there was a problem when merging at some point (that loop is actually written twice).
Regarding the next two changes: I multiplied the constraints by a filter, because sometimes, the Keccak STARK can be empty, and that made the constraints fail (as padding rows don't respect those constraints) .